src/Entities/Entity.cpp

```diff
@@ -2285,3 +2285,103 @@ void cEntity::BroadcastLeashedMobs()
 		}
 	}
 }
+
+
+
+
+
+void cEntity::BroadcastDeathMessage(TakeDamageInfo & a_TDI)
+{
+	cPluginManager * PluginManager = cRoot::Get()->GetPluginManager();
+
+	AString Name;
+	if (IsPlayer())
+	{
+		cPlayer * Player = static_cast<cPlayer *>(this);
+		Name = Player->GetName();
+	}
+	else if (IsMob())
+	{
+		cMonster * Monster = static_cast<cMonster *>(this);
+		if (Monster->HasCustomName())
+		{
+			Name = Monster->GetCustomName();
+		}
+		else
+		{
+			// Tamed ocelots are really cats in vanilla.
+			if (Monster->IsTame() && (Monster->GetClass() == AString("cOcelot")))
+			{
+				Name = "Cat";
+			}
+			else
+			{
+				Name = Monster->GetClass();
+				Name.erase(Name.begin());  // Erase the 'c' of the class (e.g. "cWitch" -> "Witch")
+			}
```

Comment:
```suggestion
	Name = NamespaceSerializer::From(Monster.GetMobType());
```

you need to `#include "WorldStorage/NamespaceSerializer.h`

Reply:
Oh - wait... this will give you the namespaced name. Not the prettified name.

Reply:
Maybe a TODO to refactor this logic somewhere reusable later on is an alternative solution?

Reply:
My suggestion would be:

NamespaceSerializer::Prettify() that crates a human-friendly version from the namespaced string:

So `snow_golem` becomes `Snow Golem`

Reply:
But we can do this later.
@TheHyper45 - your call

Reply:
OK, Let's do it later.

Reply:
OK, I wrote the NamespaceSerializer::Prettify() method but I think I should put it in another pull request.